crates/sui-core/src/module_cache_gauge.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use move_bytecode_utils::module_cache::SyncModuleCache;
+use move_core_types::resolver::ModuleResolver;
+use prometheus::{
+    core::{Collector, Desc},
+    labels,
+    proto::{Gauge, Metric, MetricFamily, MetricType},
+};
+
+/// Holds the module cache to collect its size and pass it to Prometheus, for monitoring in Grafana.
+pub struct ModuleCacheGauge<R: ModuleResolver> {
+    desc: Desc,
+    _module_cache: Arc<SyncModuleCache<R>>,
+}
+
+impl<R: ModuleResolver> ModuleCacheGauge<R> {
+    pub fn new(module_cache: Arc<SyncModuleCache<R>>) -> Self {
+        Self {
+            desc: Desc::new(
+                "module_cache_size".into(),
+                "Number of compiled move modules in the authority's cache.".into(),
+                /* variable_labels */ vec![],
+                /* const_labels */ labels! {},
+            )
+            .unwrap(),
+            _module_cache: module_cache,
+        }
+    }
+
+    fn metric(&self) -> Metric {
+        let mut m = Metric::default();
+        let mut gauge = Gauge::default();
+        // gauge.set_value(self.module_cache.len().into_f64());
+        gauge.set_value(42.);
```

Comment:
This seems like test code and the commented out code sounds like what we really want?   However the whole function I don't get, because one typically wants to keep the gauge as part of the struct and not create a new one every time.

Reply:
Yes, this code is waiting on a change to Move to land (now landed) to expose the len function that the commented out part calls -- that's why this PR is in Draft mode still!

Re: the `Gauge` here and the fact that it is being re-created, it's different from the other Gauges in `AuthoriryMetric` in that it's a protobuf model for a gauge (rather than the gauge that you set a value on usually in Prometheus APIs).  This one is intended to be re-collected each time (if you look at how those other gauges implement the `Collector` trait, they work the same way and create a new `prometheus::proto::Gauge` for every call to `collect`.